MobiFlight/Joysticks/Joystick.cs

```diff
@@ -216,32 +246,33 @@ public List<ListItem<IBaseDevice>> GetAvailableDevicesAsListItems()
 
         private List<JoystickDevice> GetButtonsSorted()
         {
-            var buttons = Buttons.ToArray().ToList();
-            buttons.Sort(SortByPositionInDefintion);
-            return Buttons;
+            return Buttons.OrderBy(button => GetIndexForJoystickDevice(button)).ToList();
         }
 
         private List<JoystickDevice> GetAxisSorted()
         {
-            var axes = Axes.ToArray().ToList();
-            Axes.Sort(SortByPositionInDefintion);
-
-            return Axes;
+            return Axes.OrderBy(axis => GetIndexForJoystickDevice(axis)).ToList();
         }
 
-        public int GetIndexForKey(string key)
+        public virtual int GetIndexForKey(string key)
         {
-            return Definition?.Inputs?.FindIndex(input => input.Name == key) ?? 0;
+            // -2 = definition not found / no inputs in definition
+            // -1 = index not found
+            // 0 or greater: valid index
+            return Definition?.Inputs?.FindIndex(input => input.Name == key) ?? -2;
```

Comment:
Documented some outputs and changed the return value in case of null pointers from a valid index to something the caller can distinguish.
Added virtual keyword because this is something that derived classes may want to override: the naming scheme of derived classes is often not based on DirectInput numbers, whereas input.Name is always based on the "Button <ID>" scheme.